libyul/AsmAnalysis.cpp

```diff
@@ -382,11 +382,25 @@ size_t AsmAnalyzer::operator()(FunctionCall const& _funCall)
 	else
 	{
 		if (!validateInstructions(_funCall))
-			m_errorReporter.declarationError(
-				4619_error,
-				nativeLocationOf(_funCall.functionName),
-				fmt::format("Function \"{}\" not found.", resolveFunctionName(_funCall.functionName, m_dialect))
-			);
+		{
+			auto const functionName = resolveFunctionName(_funCall.functionName, m_dialect);
+
+			if (functionName == "auxdataloadn" || functionName == "eofcreate" || functionName == "returncontract")
```

Comment:
A more general way to check this would be:
```c++
if (resolveBuiltinFunction(
	funCall.functionName,
	EVMDialect::strictAssemblyForEVMObjects(EVMVersion::prague(), 1)
))
```

And that also with an assert that EOF version <= 1 and EVM version <= Prague.

And a comment that `strictAssemblyForEVMObjects()` is expected to be the dialect that contains all the possible builtins. Or maybe we should just check both it and `strictAssemblyForEVM()` to be safe.

Reply:
Also, I think it would fit better in `validateInstructions()` (the overload that takes a string). Just before `return false`, when we know we're not dealing with an instruction.

Reply:
Regarding using `resolveBuiltinFunction` it would be really good but unfortunately `funCall.functionName` in `Identifier` not `BuiltinName` which means `resolveBuiltinFunction` will return always `nullptr`

Reply:
I went trough the `validateInstructions` and change there accordingly.

Reply:
The way you did it in the end looks good so marking as resolved.